app/repositories/onlinetesting/Phase2TestRepository.scala

```diff
@@ -39,6 +39,8 @@ trait Phase2TestRepository extends OnlineTestRepository with Phase2TestConcern {
 
   def getTestGroup(applicationId: String): Future[Option[Phase2TestGroup]]
 
+  def getTestGroupByUserId(userId: String): Future[Option[Phase2TestGroup]]
```

Comment:
We could probably move this and the implementation up to the OnlineTestRepository as it is generic enough.